cmd/tsbs_generate_queries/databases/mongo/devops-naive.go

```diff
@@ -42,43 +44,46 @@ func (d *NaiveDevops) GroupByTime(qi query.Query, nHosts, numMetrics int, timeRa
 	metrics, err := devops.GetCPUMetricsSlice(numMetrics)
 	panicIfErr(err)
 
-	bucketNano := time.Minute.Nanoseconds()
 	pipelineQuery := []bson.M{
 		{
-			"$match": map[string]interface{}{
+			"$match": bson.M{
 				"measurement": "cpu",
-				"timestamp_ns": map[string]interface{}{
-					"$gte": interval.StartUnixNano(),
-					"$lt":  interval.EndUnixNano(),
+				"time": bson.M{
+					"$gte": interval.Start(),
+					"$lt":  interval.End(),
 				},
-				"tags.hostname": map[string]interface{}{
+				"tags.hostname": bson.M{
 					"$in": hostnames,
 				},
 			},
 		},
 		{
-			"$project": map[string]interface{}{
+			"$project": bson.M{
 				"_id": 0,
-				"time_bucket": map[string]interface{}{
-					"$subtract": []interface{}{
-						"$timestamp_ns",
-						map[string]interface{}{"$mod": []interface{}{"$timestamp_ns", bucketNano}},
+				"time_bucket": bson.M{
+					"$dateSubtract": bson.M{
+						"startDate": "$time",
+						"unit":      "second",
+						"amount":    bson.M{"$second": bson.M{"date": "$time"}},
 					},
 				},
-
-				"fields": 1,
 			},
 		},
 	}
 
+	projectMap := pipelineQuery[1]["$project"].(bson.M)
+	for _, metric := range metrics {
+		projectMap[metric] = 1
+	}
```

Comment:
'metrics' is a slice of the stored CPU metrics, so if 'numMetrics' is 1, the pipeline $project will look something like this:
`{$project: {_id: 0, time_bucket: {...}, usage_user: 1 }}`

Reply:
Cool, if all this is already handled by our dependency analysis I'll go ahead and remove these projects!